cmake/modules/CheckCompiler.cmake

```diff
@@ -195,5 +195,6 @@ int main() {}
 #---Print the final compiler flags--------------------------------------------------------------------
 message(STATUS "ROOT Platform: ${ROOT_PLATFORM}")
 message(STATUS "ROOT Architecture: ${ROOT_ARCHITECTURE}")
-message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
-message(STATUS "Compiler Flags: ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}}")
+string(TOUPPER ${CMAKE_BUILD_TYPE} uppercase_CMAKE_BUILD_TYPE)
+message(STATUS "Build Type: ${CMAKE_BUILD_TYPE} (flags = '${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}}')")
+message(STATUS "Compiler Flags: ${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}} ${CMAKE_CXX_FLAGS}")
```

Comment:
Is `${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}}` really coming before `${CMAKE_CXX_FLAGS}` during the build? I thought it was the reverse...

Reply:
You are right. The wrong order is from the old PR where I wanted to change the order of the compile flags.
Will be fixed soon.